crates/ide-assists/src/handlers/replace_derive_with_manual_impl.rs

```diff
@@ -936,6 +936,37 @@ impl PartialEq for Foo {
         )
     }
 
+    #[test]
+    fn add_custom_impl_partial_eq_tuple_enum() {
+        check_assist(
+            replace_derive_with_manual_impl,
+            r#"
+//- minicore: eq, derive
+#[derive(Partial$0Eq)]
+enum Foo {
+    Bar(String),
+    Baz(i32),
+}
+"#,
+            r#"
+enum Foo {
+    Bar(String),
+    Baz(i32),
+}
+
+impl PartialEq for Foo {
+    $0fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Bar(l0), Self::Bar(r0)) => l0 == r0,
+            (Self::Baz(l0), Self::Baz(r0)) => l0 == r0,
+            _ => core::mem::discriminant(self) == core::mem::discriminant(other),
```

Comment:
While this is not wrong at all, I'd say `_ => false` would be nicer as at this point we've already gone through every case where the variants of `self` and `other` match. I think we can generate such fallback case when `n_cases == arms.len()` (and maybe `n_cases > 1` too)?

Reply:
@lowr Thanks a lot! I've adjusted the code accordingly.